### PR TITLE
#24 PHP 8 support

### LINKED ISSUE #24: PHP 8 support
Can we add PHP 8 in composer requirements or some code should be modified?

### PR DESCRIPTION
**Pull request by tigokr**
- php8 support
- php >=7.3 dropped
- phpstan 8
- cs fix

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,19 +18,17 @@
     }
   ],
   "require": {
-    "php": "^7.1",
```
Comment (stepanets) [inline composer.json:21]: In  arrayee you did so: `"php": ">=7.3",`

Comment (tigokr) [reply composer.json:21]: gotcha

Comment (tigokr) [reply composer.json:21]: fixed

File: src/Ratio.php [hunk 2]
```diff
@@ -16,7 +12,7 @@ interface Ratio
     /**
      * @param Currency $base
      * @param Currency $minor
-     * @return Numeric
```
Comment (stepanets) [inline src/Ratio.php:19]: what's wrong with it?

File: src/Operations/RateOf.php [hunk 3]
```diff
@@ -49,6 +46,6 @@ public function asNumber(): string
             $this->minor->currency()
         ))->asString();
 
-        return bcdiv($this->base->amount(), $this->minor->amount(), $this->scale);
+        return bcdiv($this->base->amount(), $this->minor->amount(), $this->scale) ?? '0';
```
Comment (stepanets) [inline src/Operations/RateOf.php:49]: is it possible for `bcdiv` to return `null`

File: src/Operations/Divided.php [hunk 4]
```diff
@@ -64,7 +61,7 @@ public function __construct(Money $money, string $denominator, int $scale)
      */
     public function amount(): string
     {
-        return bcdiv($this->money->amount(), $this->denominator, $this->scale);
+        return bcdiv($this->money->amount(), $this->denominator, $this->scale) ?? '0';
```
Comment (stepanets) [inline src/Operations/Divided.php:64]: is it possible for `bcdiv` to return `null`?

Comment (tigokr) [reply src/Operations/Divided.php:64]: i don't sure but phpstorm bindings tell it can

File: src/Ratio.php (hunk 2 above)
Comment (tigokr) [reply src/Ratio.php:19]: numeric is a name of pseudo type and phpstan warn about it

File: src/Operations/Divided.php (hunk 4 above)
Comment (stepanets) [reply src/Operations/Divided.php:64]: `null` will be returned if denominator is zero, we should add checks

Comment (stepanets) [reply src/Operations/Divided.php:64]: oh! I see these checks in constructor

**Review by stepanets — APPROVED** [review]

**Pull request merged**